utils/usePlatformDetection.ts

```diff
@@ -0,0 +1,16 @@
+import { useRouter } from 'next/router';
+import { useSetAtom } from 'jotai';
+import { useEffect } from 'react';
+import { currentPlatformAtom } from 'components/basecomponents/Kiosk/atoms';
+import { Platform } from '@types';
+
+export const usePlatformDetection = (): void => {
+  const router = useRouter();
+  const setCurrentPlatform = useSetAtom(currentPlatformAtom);
```

Comment:
This is still more complicated than it needs to be :) Consider

```ts
export const usePlatform = (): { currentPlatform: Platform } => {
  const router = useRouter();
  return { currentPlatform: router.asPath.includes('kiosk') ? Platform.KIOSK : Platform.WEB };
};
```

and then everywhere just 
```ts
  const { currentPlatform } = usePlatform();
```

Reply:
Updated

Reply:
Thanks! Approved